[SYCL] Handle KernelNameType templated using template template type with enum template argument

elizabethandrews opened the pull request:
Add support for KernelNameType with is a template specialization class with template template arguments which in turn is templated using enum.

Signed-off-by: Elizabeth Andrews <elizabeth.andrews@intel.com>

erichkeane commented:
@premanandrao please take a look for me.

premanandrao commented on clang/lib/Sema/SemaSYCL.cpp:
```diff
@@ -1831,6 +1831,37 @@ void SYCLIntegrationHeader::emitForwardClassDecls(
+        TemplateParameterList *TemplateParams = TD->getTemplateParameters();
+        for (NamedDecl *P : *TemplateParams) {
+          // If template template paramter type has an enum value template
+          // paramter, forward declaration of enum type is required. Only enum
```
Minor: typo

elizabethandrews commented:
@bader I don't think this Jenkins issue is related to my patch.

bader commented:
> @bader I don't think this Jenkins issue is related to my patch.

Okay. Do you know who is working on addressing these issues?

elizabethandrews commented:
> > @bader I don't think this Jenkins issue is related to my patch.
> 
> Okay. Do you know who is working on addressing these issues?

No I do not. I see it is failing for other PRs too. I'm not sure how to investigate it.
